### DISCUSSION TITLE
Consider splitting lexicon schemas into their own repo?

### DISCUSSION THREAD

**Discussion by DavidBuchanan314** [bluesky-social/atproto#1551] [General Discussion]
This might be more trouble than it's worth, but it would certainly make *my* life marginally easier if the `lexicons/` directory was its own repo, included as a git submodule. That way, I can more conveniently include it in my own codebase(s) that rely on the lexicon schemas, without having to pull the whole atproto repo.

**Comment by dholms** [comment]
Yeah +1

We've discussed this & I'd like to do so

**Comment by bnewbold** [comment]
I don't love the sub-module developer experience, though I wouldn't block/veto this.

In the longer run we may end up with a standard way to authoritatively resolve/enumerate and fetch Lexicon schema files as part of the protocol itself.

Is there a significant benefit to using a sub-module over just copying the JSON files directly in to a directory in the repo, instead of using submodules? Eg, using a Makefile command or shell script to fetch from "upstream" git repo. This provides a full Lexicon diff locally in the repo, as part of a PR, which can make reviewing history easier.
